Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -254,84 +253,90 @@ function xASL_wrp_ResampleASL(x)
 				% Number of total volumes cannot be divided by blockSize
 				error(['Total number of volumes ' xASL_num2str(size(ASL_im,4)) ' cannot be composed of blocks of size ' xASL_num2str(blockSize)]);
 			else
-				PWI = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
+				PWI4D = zeros(size(ASL_im,1), size(ASL_im,2), size(ASL_im,3), blockSize); % preallocate PWI
 				for iBlock = 1:blockSize
-					PWI(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions
+					PWI4D(:,:,:,iBlock) = xASL_stat_MeanNan(ASL_im(:,:,:,iBlock:blockSize:end), 4); % Averaged PWI4D across repetitions
 				end
 			end
         else % if Hadamard Block size == unique PLDs * number of multi-TEs -> a single repetitions -> we don't want to average
-            PWI = ASL_im;
+            PWI4D = ASL_im;
         end
-
-        % Save PWI4D
-        fprintf('%s\n', PathPWI4D{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, PWI, 32, false);
         
         % Create single PWI for further steps in ASL module
-        PWIsingle = xASL_stat_MeanNan(PWI(:,:,:,1:x.Q.NumberEchoTimes:end),4); % Average across PLDs from each first TE
+        PWI3D = xASL_stat_MeanNan(PWI4D(:,:,:,1:x.Q.NumberEchoTimes:end),4); % Average across PLDs from each first TE

```

Comment:
This should then be PWI = PWI3D( .... but I don't know yet which volumes are which TE/PLD/labdur (see above)

Reply:
We've discussed this, right?
First TEs, then PLDs/Labdur. It's coded like that in the import to reorder correctly accoridng to the acquisition order where TEs are always first.

We can change that in future, but so far all datasets we had have TE acquisitions together.

Reply:
Yes. all good. not relevant anymore. I'm resolving.